temporal-test-server/src/main/java/io/temporal/internal/testservice/TestVisibilityStoreImpl.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.internal.testservice;
+
+import io.grpc.Status;
+import io.temporal.api.enums.v1.IndexedValueType;
+import java.util.HashMap;
+import java.util.Map;
+
+public class TestVisibilityStoreImpl implements TestVisibilityStore {
+
+  private static final String TEST_KEY_STRING = "CustomStringField";
+  private static final String TEST_KEY_INTEGER = "CustomIntField";
+  private static final String TEST_KEY_DATE_TIME = "CustomDatetimeField";
+  private static final String TEST_KEY_DOUBLE = "CustomDoubleField";
+  private static final String TEST_KEY_BOOL = "CustomBoolField";
+
+  private final Map<String, IndexedValueType> searchAttributes =
+      new HashMap<String, IndexedValueType>() {
```

Comment:
Worthwhile to throw a lock around access to this? Or make it a synchronized map?